src/btree/bt_cursor.c

```diff
@@ -948,7 +948,7 @@ __cursor_truncate(WT_SESSION_IMPL *session,
 	} else {
 		do {
 			WT_RET(__wt_btcur_remove(start));
-			for (;;) {
+			for (ret = 0;;) {
```

Comment:
This is subtle: without it, we ended up at the end of the scan (`start` == `stop`) and `ret == WT_RESTART`, and got stuck in an infinite loop.  Only with column stores, and I'm not sure exactly where the `WT_RESTART` was coming from, but it's clearly a weakness of these loops.